components/canvas/canvas_paint_thread.rs

```diff
@@ -30,7 +30,7 @@ pub enum ImageUpdate {
 }
 
 pub trait WebrenderApi {
-    fn generate_key(&self) -> Result<ImageKey, ()>;
+    fn generate_key(&self) -> Option<ImageKey>;
```

Comment:
```suggestion
    /// Attempt to generate an [`ImageKey`], returning `None` in case of failure.
    fn generate_key(&self) -> Option<ImageKey>;
```